### PR TITLE
Add MCP server with five CRAIC tools

### LINKED ISSUE #4: CRAIC MCP server (FastMCP)
## Description

Implement the MCP server using FastMCP that exposes the five CRAIC tools. This is the core component — Claude Code spawns it via stdio.

Ref: [design doc — MCP Tool Interface](docs/plans/2026-03-04-craic-mvp-poc-design.md#mcp-tool-interface)

## Deliverables

- `server/craic_mcp/server.py` — FastMCP server
- Five tools:
  - `craic_query` — search local store (and team store when #6 lands), merge/dedup, rank by relevance * confidence
  - `craic_propose` — create knowledge unit in local store (and team store if generic)
  - `craic_confirm` — increment confirmation count, recalculate confidence
  - `craic_flag` — flag unit as stale/incorrect/duplicate, decrement confidence
  - `craic_reflect` — analyse session context, return candidate knowledge units
- Runnable via `uv run craic-mcp-server`
- Integration tests

## Depends on

- #2 (knowledge unit model)
- #3 (local store)

### PR DESCRIPTION
**Pull request by peteski22**
## Summary

- Implement the MCP server using the official `mcp` SDK, exposing five tools via stdio transport: `craic_query`, `craic_propose`, `craic_confirm`, `craic_flag`, and `craic_reflect`
- Thread-safe lazy store initialisation with double-checked locking and `atexit` cleanup
- Input validation at the tool boundary: domain tag stripping/filtering, blank field rejection, limit bounds checking
- `craic_reflect` registered as a stub with `status` field (real session mining deferred to #9)
- 25 integration tests covering all tools, validation errors, and an end-to-end lifecycle test

## Test plan

- [x] All 25 server tests pass (`make test` — 132 total across server + team-api)
- [x] Lint, format, and typecheck pass (`make lint`)
- [x] End-to-end lifecycle test: propose → query → confirm → flag with confidence tracking
- [x] Validation edge cases: empty domains, whitespace-only domains, blank fields, zero/negative limit, missing units, invalid flag reasons

Closes #4

### CODE REVIEW

File: server/craic_mcp/server.py [hunk 1]
```diff
@@ -1,9 +1,228 @@
...
+
+    Returns:
+        Dict with ``results`` (list of knowledge unit dicts) and ``source``,
+        or ``error`` if inputs are invalid.
+    """
+    cleaned = [d.strip() for d in domain if d.strip()]
+    if not cleaned:
+        return {"error": "At least one non-empty domain tag is required."}
+    if limit < 1:
+        return {"error": "limit must be a positive integer."}
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py:80]: Fixed. Added `_MAX_QUERY_LIMIT = 50` upper bound with validation and test.

File: server/craic_mcp/server.py [hunk 2]
```diff
@@ -1,9 +1,228 @@
...
+    if not summary.strip() or not detail.strip() or not action.strip():
+        return {"error": "summary, detail, and action must be non-blank."}
+    cleaned_domain = [d.strip() for d in domain if d.strip()]
+    if not cleaned_domain:
+        return {"error": "At least one non-empty domain tag is required."}
+    store = _get_store()
+    context = Context(
+        languages=[language] if language else [],
+        frameworks=[framework] if framework else [],
+        pattern=pattern,
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py]: Fixed. `language`, `framework`, and `pattern` are now stripped; whitespace-only values are treated as absent. Added three tests covering strip and blank-as-None behaviour.

File: server/craic_mcp/server.py [hunk 3]
```diff
@@ -1,9 +1,228 @@
...
+    """
+    store = _get_store()
+    unit = store.get(unit_id)
+    if unit is None:
+        return {"error": f"Knowledge unit not found: {unit_id}"}
+    try:
+        flag_reason = FlagReason(reason)
+    except ValueError:
+        valid = ", ".join(r.value for r in FlagReason)
+        return {"error": f"Invalid reason: {reason}. Must be one of: {valid}."}
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py:193]: Fixed. `reason` is now stripped and lowercased before enum conversion. Added two tests for whitespace and case normalisation.

File: server/craic_mcp/server.py [hunk 4]
```diff
@@ -1,9 +1,228 @@
...
+    return _store
+
+
+def _close_store() -> None:
+    """Close the store if open."""
+    global _store  # noqa: PLW0603
+    with _store_lock:
+        if _store is not None:
+            _store.close()
+            _store = None
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py]: Fixed. Switched to `threading.local()` so each thread gets its own `LocalStore` instance. Removed the lock since thread-local storage is inherently thread-safe per-thread.

File: server/craic_mcp/server.py [hunk 5]
```diff
@@ -1,9 +1,228 @@
...
+
+    Thread-safe: uses double-checked locking to ensure a single instance.
+    """
+    global _store  # noqa: PLW0603
+    if _store is None:
+        with _store_lock:
+            if _store is None:
+                db_path_str = os.environ.get("CRAIC_LOCAL_DB_PATH")
+                db_path = Path(db_path_str) if db_path_str else None
+                _store = LocalStore(db_path=db_path)
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py]: Fixed. The `threading.local()` change makes this moot — `_get_store()` now returns a local variable rather than a global, so the type narrowing is straightforward.

File: server/craic_mcp/server.py [hunk 6]
```diff
@@ -1,9 +1,228 @@
...
+        or ``error`` if inputs are invalid.
+    """
+    cleaned = [d.strip() for d in domain if d.strip()]
+    if not cleaned:
+        return {"error": "At least one non-empty domain tag is required."}
+    if limit < 1:
+        return {"error": "limit must be a positive integer."}
+    store = _get_store()
+    results = store.query(cleaned, language=language, framework=framework, limit=limit)
+    return {
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py:85]: Fixed. Added `_MAX_QUERY_LIMIT = 50` upper bound with validation and test.

File: server/craic_mcp/server.py [hunk 7]
```diff
@@ -1,9 +1,228 @@
...
+        flag_reason = FlagReason(reason)
+    except ValueError:
+        valid = ", ".join(r.value for r in FlagReason)
+        return {"error": f"Invalid reason: {reason}. Must be one of: {valid}."}
+    flagged = apply_flag(unit, flag_reason)
+    store.update(flagged)
+    return {
+        "id": flagged.id,
+        "new_confidence": flagged.evidence.confidence,
+        "message": f"Knowledge unit {flagged.id} flagged as {reason}.",
```
[the remark this answers is not in the document]
Comment (peteski22) [reply server/craic_mcp/server.py:199]: Fixed. `reason` is stripped and lowercased before enum conversion. Two tests added.
